plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -502,41 +503,41 @@ public void testMySqlDatetimeType(ZoneId sessionZone)
                 // TODO time gaps do not round-trip (https://github.com/trinodb/trino/issues/6910)
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:13:42.000000'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:13:42.000'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '2018-04-01 02:13:55.123000'", TIMESTAMP_MILLIS, "TIMESTAMP '2018-04-01 03:13:55.123'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '2018-03-25 03:17:17.000000'", TIMESTAMP_MILLIS, "TIMESTAMP '2018-03-25 03:17:17.000'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1986-01-01 00:13:07.000000'", TIMESTAMP_MILLIS, "TIMESTAMP '1986-01-01 00:13:07.000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 01:13:42.000000'", TIMESTAMP_MICROS, "TIMESTAMP '1970-01-01 01:13:42.000000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '2018-04-01 03:13:55.123000'", TIMESTAMP_MICROS, "TIMESTAMP '2018-04-01 03:13:55.123000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '2018-03-25 03:17:17.000000'", TIMESTAMP_MICROS, "TIMESTAMP '2018-03-25 03:17:17.000000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1986-01-01 00:13:07.000000'", TIMESTAMP_MICROS, "TIMESTAMP '1986-01-01 00:13:07.000000'")
 
                 // test arbitrary time for all supported precisions
-                .addRoundTrip("datetime(0)", "TIMESTAMP '1970-01-01 00:00:01'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.000'")
-                .addRoundTrip("datetime(1)", "TIMESTAMP '1970-01-01 00:00:01.1'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.100'")
-                .addRoundTrip("datetime(2)", "TIMESTAMP '1970-01-01 00:00:01.12'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.120'")
-                .addRoundTrip("datetime(3)", "TIMESTAMP '1970-01-01 00:00:01.123'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("datetime(4)", "TIMESTAMP '1970-01-01 00:00:01.1234'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("datetime(5)", "TIMESTAMP '1970-01-01 00:00:01.12345'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.123456'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
+                .addRoundTrip("datetime(0)", "TIMESTAMP '1970-01-01 01:00:01'", createTimestampType(0), "TIMESTAMP '1970-01-01 01:00:01'")
+                .addRoundTrip("datetime(1)", "TIMESTAMP '1970-01-01 01:00:01.1'", createTimestampType(1), "TIMESTAMP '1970-01-01 01:00:01.1'")
+                .addRoundTrip("datetime(2)", "TIMESTAMP '1970-01-01 01:00:01.12'", createTimestampType(2), "TIMESTAMP '1970-01-01 01:00:01.12'")
+                .addRoundTrip("datetime(3)", "TIMESTAMP '1970-01-01 01:00:01.123'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
```

Comment:
nit: Use `createTimestampType(3)` for consistency.

Reply:
Upgrading the driver to `8.0.22` seems to fix all roundtrip issues  for `timestamp` and `datetime`